### PR TITLE
Assignments day filter

### PR DESCRIPTION
**Pull request by thecalcc**
SDBELGA-819

TODO: 
- [x] Check counter of list items
- [x] Check calendar 'Clear' button alignment
- [x] Check filter persistence when switching between Desk/My assignments view with Gideon
- [x] Go over all usages in superdesk-cc [client-core pr](https://github.com/superdesk/superdesk-client-core/pull/4564)
- [x] Implement day filter persistence

### CODE REVIEW

**Comment by thecalcc** [comment]
@tomaskikutis ignore the commented out code when doing a review now, as this is something to be discussed with Gideon

File: client/components/UI/List/Header.tsx [hunk 1]
```diff
@@ -1,13 +1,19 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import classNames from 'classnames';
 
+interface IProps {
+    title?: string;
+    id?: string;
+    marginTop?: boolean;
+    marginBottom?: boolean;
+    children?: JSX.Element | Array<JSX.Element>;
```
Comment (tomaskikutis) [inline client/components/UI/List/Header.tsx]: wouldn't `children?: React.ReactNode` work?

File: client/components/Assignments/FiltersBar.tsx [hunk 2]
```diff
@@ -3,26 +3,46 @@ import PropTypes from 'prop-types';
...
 
-import {SubNav} from 'superdesk-ui-framework/react';
+import {DatePicker, SubNav} from 'superdesk-ui-framework/react';
 import {StretchBar, Spacer} from '../UI/SubNav';
 import {Checkbox} from '../UI/Form';
 import {OrderFieldInput} from '../OrderBar';
 import {DesksSubnavDropdown} from './DesksSubNavDropDown';
+import {appConfig} from 'superdesk-core/scripts/appConfig';
 
+interface IProps {
```
Comment (tomaskikutis) [inline client/components/Assignments/FiltersBar.tsx:10]: if you're adding `IProps`, drop prop-types.

File: client/components/Assignments/FiltersBar.tsx [hunk 3]
```diff
@@ -3,26 +3,46 @@ import PropTypes from 'prop-types';
...
+    showAllDeskOption?: boolean;
+    showDeskAssignmentView?: boolean;
+}
+
+const DAYS_AHEAD = 6;
 
 export const FiltersBar = ({
-    filterBy,
-    orderByField,
+    filterBy = 'Desk',
```
Comment (tomaskikutis) [inline client/components/Assignments/FiltersBar.tsx:27]: double-check that react default props use the same logic to apply a default as default function arguments. That is if value is undefined - a default gets applied. If `null` - then not.

File: client/components/Assignments/AssignmentGroupList.tsx [hunk 4]
```diff
@@ -248,9 +254,9 @@ class AssignmentGroupListComponent extends React.Component {
                             <div className="sd-list-header__number sd-flex-grow">
                                 <span className="a11y-only">{gettext(
                                     'Number of Assignments: ',
-                                    {count: totalCount}
+                                    {count: (filteredAssignments?.length ?? 0)}
```
Comment (tomaskikutis) [inline client/components/Assignments/AssignmentGroupList.tsx:297]: would be good to update this to use placeholder based translation. It should be possible to pass a react component like `gettext('Number of Assignments: {{n}}', {n: () => <div>5</div>})`

File: client/components/Assignments/AssignmentGroupList.tsx [hunk 5]
```diff
@@ -283,8 +289,8 @@ class AssignmentGroupListComponent extends React.Component {
                         <ListItemLoader />
                     )}
                     {isLoading !== true && (
-                        get(assignments, 'length', 0) > 0 ? (
-                            assignments.map((assignment, index) => this.rowRenderer(index))
+                        (filteredAssignments?.length ?? 0) > 0 ? (
+                            filteredAssignments.map((assignment, index) => this.rowRenderer(index))
```
Comment (tomaskikutis) [inline client/components/Assignments/AssignmentGroupList.tsx]: if first argument isn't being used, we should rename it to `_`

File: client/apps/Assignments/AssignmentList.tsx [hunk 6]
```diff
@@ -79,6 +79,7 @@ AssignmentListContainer.defaultProps = {
 };
 
 const mapStateToProps = (state) => ({
+    dayField: selectors.getDayField(state),
```
Comment (tomaskikutis) [inline client/apps/Assignments/AssignmentList.tsx]: is this used?

File: client/components/Assignments/AssignmentGroupList.tsx (hunk 4 above)
Comment (thecalcc) [reply client/components/Assignments/AssignmentGroupList.tsx:297]: We can create a ticker for this if you want to update it

File: client/apps/Assignments/AssignmentList.tsx (hunk 6 above)
Comment (thecalcc) [reply client/apps/Assignments/AssignmentList.tsx]: nope, removed

File: client/components/Assignments/FiltersBar.tsx (hunk 3 above)
Comment (thecalcc) [reply client/components/Assignments/FiltersBar.tsx:27]: I don't think this is React specific. It's a function like any other one in javascript. In any case I tested how JS functions work with default arguments, so the same:

image

Comment (thecalcc) [reply client/components/Assignments/FiltersBar.tsx:27]: I've also tested it with this current component. When I pass `undefined` to `orderByField` argument, it falls back. When I pass `null` to it, there's no fallback.

**Comment by thecalcc** [comment]
demo:

https://github.com/superdesk/superdesk-planning/assets/45574768/75b601ca-4b9b-443b-83f2-6b7ad6a4ccd1

File: client/components/Assignments/FiltersBar.tsx (hunk 3 above)
Comment (tomaskikutis) [reply client/components/Assignments/FiltersBar.tsx:27]: > I don't think this is React specific.

Of course logic on how `defaultProps` work is react specific. `defaultProps` don't exist without react. But all good if it works the same as default values in native JS functions.

File: client/components/Assignments/AssignmentGroupList.tsx (hunk 4 above)
Comment (tomaskikutis) [reply client/components/Assignments/AssignmentGroupList.tsx:297]: it's 2 min maybe you can update without a ticket? :)

**Review by tomaskikutis — APPROVED** [review]

**Pull request merged**